Jenkinsfile

```diff
@@ -632,6 +638,72 @@ pipeline {
       }
     }
 
+    stage('Deploy Preview') {
+      when {
+        not {
+          branch 'PR-*'
+        }
+      }
+      steps {
+        setGitHubBuildStatus('nuxeo/preview', 'Deploy nuxeo Preview', 'PENDING')
+        container('maven') {
+          dir('ci/helm/preview') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute environment variables in chart values
+            sh """
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              boolean nsExist = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+              String noCommentOpt = '';
+              if (nsExist) {
+                noCommentOpt = '--no-comment'
+              } else {
+                sh "kubectl create namespace ${PREVIEW_NAMESPACE}"
+              }
+              sh "kubectl create secret generic preview-docker-cfg --namespace=${PREVIEW_NAMESPACE} --from-file=.dockerconfigjson=/home/jenkins/.docker/config.json --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"
+              // third build and deploy the chart
+              // reference branch have a different preview command. The -n name provided must be lowercase.
+              previewCommand = "jx step helm install --namespace ${PREVIEW_NAMESPACE} -n ${APP_NAME}-${PREVIEW_NAMESPACE.toLowerCase()} ."
+              if (isPullRequest()) {
+                // waiting for https://github.com/jenkins-x/jx/issues/5797 to be fixed in order to remove --source-url
+                previewCommand = "jx preview --namespace ${PREVIEW_NAMESPACE} --source-url=https://github.com/nuxeo/nuxeo ${noCommentOpt}"
+              }
+              sh """
+                jx step helm build
+                mkdir target && helm template . --output-dir target
+                ${previewCommand}
+              """
+              if (!isPullRequest()) {
+                // When not using jx preview, we need to expose the nuxeo url by hand
+                url = sh(returnStdout: true, script: "jx get urls -n nuxeo-pr-3856 | grep -oP https://.* | tr -d '\\n'")
```

Comment:
I guess `nuxeo-pr-3856` should be `${PREVIEW_NAMESPACE}`?

Reply:
indeed I thought I fixed this.